zachyot/zachyot/Program.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace zachyot
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+         
+            for (var number=10; number<=250000000;number++)
+               
+                for(var i=2; i<=number; i++)
+               
+                {
+                    if (Math.Pow(Sum(number), i) > number)
+                        break;
+                   
+                    else if(Math.Pow(Sum(number),i)==number)
+                        Console.WriteLine($"{number}={Sum(number)}^{i}");
+                }
+
+            Console.ReadKey();
+        }
```

Comment:
Можно было бы при проверке каждой степени не вычислять заново сумму цифр. Программа бы ускорилась.
```suggestion
            for (var number=10; number<=250000000;number++)
            {
                var s = Sum(number);
                
                for(var i=2; i<=number; i++)               
                    if (Math.Pow(s > number)
                        break;
                    else if(Math.Pow(s, i)==number)
                        Console.WriteLine($"{number}={s}^{i}");
            }

            Console.ReadKey();
        }
```
Так (на моем компьютере) ваша программа нашла нужные числа за 4 m 23 s 872 ms.
С указанной выше модернизацией она работала 1 m 39 s 235 ms. Более чем в 2 раза ускорилась!